src/App.jsx

```diff
@@ -64,6 +64,8 @@ export function App({ gmApiKey, gaTag }) {
   const [userLongitude, setUserLongitude] = useState(0);
   const [currentLocationLoaded, setCurrentLocationLoaded] = useState(false);
 
+  const slug = useRefillSpotSlug();
```

Comment:
I think this is LGTM, but could you just explain why you want to add a custom hook instead of just doing
```
        const slug = getSlug(REFILL_SPOT_ROUTE);
``` 
here and sharing the constant in both places where it's used?

Reply:
Hey, thanks for the review.

getSlug access `window` inside, which is only available on the client side. Thus, if the component code was executed on the server side, it will throw an error since `window` is not defined. Based on my understanding, this could be a problem in some frameworks like Next.js, since sometimes the component code is [executed on the server side](https://nextjs.org/docs/app/building-your-application/rendering). Thus, we normally use `useEffect` to ensure we access the window only when the component is rendered.

I was under the impression that similar things could happen to us since we were using ReactDOMServer. However, I was unfamiliar with ReactDOMServer and may have made the wrong assumption. 

I tried replacing it directly to `getSlug` as you recommended, and have not encountered an error yet. So, I will investigate further whether `useEffect` is truly needed or not for our case.

Reply:
<img width="613" alt="Screenshot 2024-09-02 at 23 04 27" src="https://github.com/user-attachments/assets/39dd3bd9-3342-44b1-8393-66899a782bee">

So it turns out that when we run `npm run build`, the code fails because `window` is not defined.

I think it is because [`ReactDOMServer` renders the component into HTML in the server](https://react.dev/reference/react-dom/server), when `window` is not defined